JetNews/app/src/main/java/com/example/jetnews/ui/home/HomeRoute.kt

```diff
@@ -210,3 +204,39 @@ private enum class HomeScreenType {
     Feed,
     ArticleDetails
 }
+
+private fun getHomeScreenType(
+    windowSize: WindowSize,
+    uiState: HomeUiState
+): HomeScreenType = when (windowSize) {
+    WindowSize.Compact,
+    WindowSize.Medium -> {
+        when (uiState) {
+            is HomeUiState.HasPosts -> {
+                if (uiState.isArticleOpen) {
+                    HomeScreenType.ArticleDetails
+                } else {
+                    HomeScreenType.Feed
+                }
+            }
+            is HomeUiState.NoPosts -> HomeScreenType.Feed
+        }
+    }
+    WindowSize.Expanded -> HomeScreenType.FeedWithArticleDetails
+}
+
+@SuppressLint("ComposableModifierFactory", "ModifierFactoryExtensionFunction")
```

Comment:
I'm not sure if this will survive with the insets changes, but I feel like we shouldn't be suppressing these lint checks if it does.

Reply:
Good call, fixed